### PR TITLE
Add gradient clipping to create_supervised_trainer()

### LINKED ISSUE #419: add gradient clipping to `create_supervised_trainer()`
It would be good to add gradient clipping to the trainers created by `create_supervised_trainer`. This is already provided by `torch.nn.utils.clip_grad_norm_`.

One possible implementation could be:

```python
import math
from torch.nn.utils import clip_grad_norm_

def create_supervised_trainer(model, optimizer, loss_fn,
                              device=None, non_blocking=False,
                              prepare_batch=_prepare_batch,
                              gradient_clip=math.inf):
    """
    Factory function for creating a trainer for supervised models.
    Args:
        model (`torch.nn.Module`): the model to train.
        optimizer (`torch.optim.Optimizer`): the optimizer to use.
        loss_fn (torch.nn loss function): the loss function to use.
        device (str, optional): device type specification (default: None).
            Applies to both model and batches.
        non_blocking (bool, optional): if True and this copy is between CPU and GPU, the copy may occur asynchronously
            with respect to the host. For other cases, this argument has no effect.
        prepare_batch (callable, optional): function that receives `batch`, `device`, `non_blocking` and outputs
            tuple of tensors `(batch_x, batch_y)`.
        gradient_clip (float, optional): value to use to clip gradients.
    Note: `engine.state.output` for this engine is the loss of the processed batch.
    Returns:
        Engine: a trainer engine with supervised update function.
    """
    if device:
        model.to(device)

    def _update(engine, batch):
        model.train()
        optimizer.zero_grad()
        x, y = prepare_batch(batch, device=device, non_blocking=non_blocking)
        y_pred = model(x)
        loss = loss_fn(y_pred, y)
        loss.backward()
        clip_grad_norm_(model.parameters(), gradient_clip)
        optimizer.step()
        return loss.item()

    return Engine(_update)
```

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Fixes #419

Added gradient clipping feature to `create_supervised_trainer()`, whatever the device or the amp mode.

### CODE REVIEW

File: ignite/contrib/metrics/cohen_kappa.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
+from typing import Callable
```
Comment (vfdev-5) [inline ignite/contrib/metrics/cohen_kappa.py:1]: This file should not be in this PR

Comment (KickItLikeShika) [reply ignite/contrib/metrics/cohen_kappa.py]: Has been fixed
